pkg/collector/corechecks/snmp/internal/report/report_bandwidth_usage.go

```diff
@@ -99,73 +101,66 @@ func (ms *MetricSender) sendBandwidthUsageMetric(symbol profiledefinition.Symbol
 	usageValue := ((octetsFloatValue * 8) / (float64(ifSpeed))) * 100.0
 
 	interfaceID := ms.deviceID + ":" + fullIndex + "." + usageName
-	err = ms.calculateRate(interfaceID, ifSpeed, usageValue, usageName, tags)
+	rate, err := ms.interfaceBandwidthState.calculateBandwidthUsageRate(interfaceID, ifSpeed, usageValue)
 	if err != nil {
 		return err
 	}
+	sample := MetricSample{
+		value:      valuestore.ResultValue{SubmissionType: profiledefinition.ProfileMetricTypeGauge, Value: rate},
+		tags:       tags,
+		symbol:     profiledefinition.SymbolConfig{Name: usageName + ".rate"},
+		forcedType: profiledefinition.ProfileMetricTypeGauge,
+		options:    profiledefinition.MetricsConfigOption{},
+	}
+	ms.sendMetric(sample)
 	return nil
 }
 
 /*
 calculateRate is responsible for checking the state for previously seen metric sample to generate the rate from.
 If the ifSpeed has changed for the interface, the rate will not be submitted (drop the previous sample)
 */
-func (ms *MetricSender) calculateRate(interfaceID string, ifSpeed uint64, usageValue float64, usageName string, tags []string) error {
-	ms.interfaceRateMap.mu.RLock()
-	interfaceRate, ok := ms.interfaceRateMap.rates[interfaceID]
-	ms.interfaceRateMap.mu.RUnlock()
-
+func (ibs *InterfaceBandwidthState) calculateBandwidthUsageRate(interfaceID string, ifSpeed uint64, usageValue float64) (float64, error) {
```

Comment:
One more thing. Currently, `InterfaceBandwidthState` struct is in `report_metrics.go` but the method `calculateBandwidthUsageRate` is in `report_bandwidth_usage.go` (in middle of other MetricSender methods).

I think this makes it a bit harder to read/follow.

What do you think about moving all the interface bandwidth usage code in a separate `interface_bandwidth_usage.go` that contains:
- InterfaceBandwidthState struct
- calculateBandwidthUsageRate method
- BandwidthUsage struct
?